src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,70 @@
+package duke.parser;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.CommandList;
+import duke.command.DoneCommand;
+import duke.command.ListTasksCommand;
+import duke.command.CreateCommand;
+import duke.command.AddCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+
+    /**
+     * Parses the user command and returns the appropriate command object.
+     * @param command A string containing user input.
+     * @return A command object.
+     * @throws DukeException throws an exception if user gave illegal input.
+     */
+    public static Command parse(String[] command) throws DukeException {
+        try {
+            CommandList commandCode = CommandList.valueOf(command[0].toUpperCase());
+            String details = "";
+
+            try {
+                details = command[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+
+            }
+
+            Task task;
+            switch (commandCode) {
+            case BYE:
+                return new ByeCommand();
+            case LIST:
+                return new ListTasksCommand();
+            case TODO:
+                task = CreateCommand.createTodo(details);
+                return AddCommand.createAddCommand(task);
+            case DEADLINE:
+                task = CreateCommand.createDeadline(details);
+                return AddCommand.createAddCommand(task);
+            case EVENT:
+                task = CreateCommand.createEvent(details);
+                return AddCommand.createAddCommand(task);
```

Comment:
I think that you can consider allowing the cases to fall through so that you would not need to duplicate the code.